build-system/tasks/helpers.js

```diff
@@ -169,7 +169,7 @@ function compile(watch, shouldMinify) {
         watch,
         extraGlobs: ['extensions/amp-viewer-integration/**/*.js'],
         compilationLevel: 'WHITESPACE_ONLY',
-        minify: false,
+        minify: shouldMinify,
```

Comment:
I just realized this change will generate `amp-viewer-host.js` and not `amp-viewer-host.max.js` when you run `gulp dist --fortesting` followed by `gulp serve --compiled`. This will break the viewer demo pages since the HTML files are hard-coded to look for `amp-viewer-host.max.js` whether or not it's a minified build.

https://github.com/ampproject/amphtml/blob/1e66423afab615a0df5d36ad43757bd29379a554/examples/viewer.html#L8
https://github.com/ampproject/amphtml/blob/5f75efe35b734a4fcf0884d373315de60cd42df9/examples/viewer-webview.html#L8
https://github.com/ampproject/amphtml/blob/5f75efe35b734a4fcf0884d373315de60cd42df9/examples/viewer-iframe-poll.html#L8

I think this can be fixed by dynamically placing the correct binary name in the viewer html files. If that's not possible, it should be possible to simply use the same binary name for both minified and unminified versions of the viewer binary.

/cc @jridgewell @choumx for comment

Reply:
Good catch. The dev server already dynamically replaces some URLs at serve-time with the [app-utils.js#replaceUrls](https://github.com/ampproject/amphtml/blob/master/build-system/app-utils.js#L64) function, so this should only need to be added to that logic.

Reply:
I added the replace logic here as part of this PR - https://github.com/ampproject/amphtml/blob/master/build-system/app-utils.js#L85

Do you still see viewer demo pages breaking?

Reply:
Pretty sure I did see them break, but it's worth double checking. Maybe the replace logic isn't being run on the viewer HTML files?

Reply:
False alarm! I just realized that the first couple of times I tried this, the page was still in the "please wait" state, which led me to believe it was using an incorrect URL. Carry on!